### PR TITLE
[WIP] HA Elastic

### PR DESCRIPTION
**Pull request by jmcarp**
Adapted from kubernetes/charts.

### CODE REVIEW

File: catalogData/elasticsearch24/3x-ha/k8s/deployment-master.json [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+              "requests": {
+                "memory": "1792M",
+                "cpu": "0"
+              },
+              "limits": {
+                "memory": "3584M",
+                "cpu": 1
+              }
+            },
+            "readinesProbe": {
```
Comment (LinuxBozo) [inline catalogData/elasticsearch24/3x-ha/k8s/deployment-master.json]: Is that a typo for readiness or is that misspelled upstream somewhere?

File: catalogData/elasticsearch24/3x-ha/k8s/statefulset-data.json [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+              "requests": {
+                "memory": "1792M",
+                "cpu": "0"
+              },
+              "limits": {
+                "memory": "3584M",
+                "cpu": 1
+              }
+            },
+            "readinesProbe": {
```
Comment (LinuxBozo) [inline catalogData/elasticsearch24/3x-ha/k8s/statefulset-data.json]: Same question here, typo?

File: catalogData/elasticsearch24/3x-ha/plan.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
+  "id": "60defe53-526c-4dd8-aeb1-226d94a7cb48",
+  "name": "3x-ha",
+  "description": "Elasticsearch instance with 3GB of RAM and 1 AWS vCPU",
```
Comment (LinuxBozo) [inline catalogData/elasticsearch24/3x-ha/plan.json:4]: Probably need a better description here.

File: catalogData/elasticsearch24/3x-ha/k8s/statefulset-master.json [hunk 4]
```diff
@@ -141,12 +142,34 @@
...
+      {
+        "metadata": {
+          "name": "data"
+        },
+        "spec": {
+          "accessModes": ["ReadWriteOnce"],
+          "storageClassName": "$storage_class",
+          "resources": {
+            "requests": {
+              "storage": "10Gi"
```
Comment (jmcarp) [inline catalogData/elasticsearch24/3x-ha/k8s/statefulset-master.json]: Not that the cost is significant, but how much disk are the masters likely to use?

File: catalogData/elasticsearch24/3x-ha/k8s/statefulset-data.json [hunk 5]
```diff
@@ -151,7 +151,7 @@
                 }
               }
             },
-            "imagePullPolicy": "Always"
+            "imagePullPolicy": "IfNotPresent"
```
Comment (jmcarp) [inline catalogData/elasticsearch24/3x-ha/k8s/statefulset-data.json]: What impact does this have on our daily rebuild? Based on https://kubernetes.io/docs/concepts/configuration/overview/#container-images, I'm thinking we'll never re-pull the image with the `2.4.4` tag, even if it's changed at the registry. Is this where we bump the image tag via environment variable like you suggested earlier?

Comment (cnelson) [reply catalogData/elasticsearch24/3x-ha/k8s/statefulset-data.json]: I think so, but doing that right is out of scope for this story IMO.

I think until we can get to the promote images via tests feature, we could ship a cronjob on the minion nodes that docker pulls our images on some interval, WDYT?

File: catalogData/elasticsearch24/3x-ha/k8s/statefulset-master.json (hunk 4 above)
Comment (cnelson) [reply catalogData/elasticsearch24/3x-ha/k8s/statefulset-master.json]: Certainly no more than our production logsearch cluster which is using a grand total of 97MB.  I updated this to use the minimum EBS size of 1GB

File: catalogData/elasticsearch24/3x-ha/plan.json (hunk 3 above)
Comment (cnelson) [reply catalogData/elasticsearch24/3x-ha/plan.json:4]: Agreed, but I think we should decide on what plans we want to offer before we write the descriptions. Thread here about deciding on plans https://gsa-tts.slack.com/conversation/C0ENP71UG/p1501115222403278

**Review by rogeruiz — APPROVED** [review]
:shipit: so we can keep plugging away on #55

**Pull request merged**